draft-ietf-quic-qlog-main-schema.md

```diff
@@ -592,68 +592,86 @@ JSON serialization:
+
+There is no requirement that events in the same trace use the same time format.
+However, using a single time format for related events can make them easier to
+analyze.
 
 Definition:
 
 ~~~ cddl
-TimeFormat = "absolute" / "delta" / "relative"
+TimeFormat = "absolute" /
+             "delta" /
+             "relative" /
+             "monotonic" .default "absolute"
 ~~~
 {: #time-format-def title="TimeFormat definition"}
 
-* Absolute: Include the full absolute timestamp with each event. This approach
-  uses the largest amount of characters. This is also the default value of the
-  "time_format" field.
-* Delta: Delta-encode each time value on the previously logged value. The first
-  event in a trace typically logs the full absolute timestamp. This approach uses
-  the least amount of characters.
-* Relative: Specify a full "reference_time" timestamp (typically this is done
-  up-front in "common_fields", see {{common-fields}}) and include only
-  relatively-encoded values based on this reference_time with each event. The
-  "reference_time" value is typically the first absolute timestamp. This approach
-  uses a medium amount of characters.
-
-The first option is good for stateless loggers, the second and third for stateful
-loggers. The third option is generally preferred, since it produces smaller files
-while being easier to reason about. An example for each option can be seen in
+absolute:
+: A full absolute timestamp. This approach uses the
+  largest amount of characters. This is the default value of the "time_format"
+  field.
+
+delta:
+: A delta-encoded value, based on the previously logged value. The first event
+  in a trace typically logs the full absolute timestamp. This approach uses the
+  least amount of characters.
+
+relative:
+: A relative timestamp based on some relative time base.  The log SHOULD specify
+  a "reference_time" timestamp, either by providing it explicitly in the
+  "reference_time" field in "common_fields" ({{common-fields}}) or logging an
+  absolute time in the first event. This approach uses a medium amount of
+  characters.
+
+monotonic:
+: A relative timestamp based on an opaque monotonically nondecreasing clock.
+  The "reference_time" field MUST NOT be applied to this format, since it based
+  on an opaque clock that cannot be interpreted beyond the process that
+  instantiated it.
+
+Events in each individual trace SHOULD be logged in strictly ascending timestamp
+order (though not necessarily absolute value, for the "delta" format). Tools CAN
+sort all events on the timestamp before processing them, though are not required
+to (as this could impose a significant processing overhead). This can be a problem
+especially for multi-threaded and/or streaming loggers, who could consider using a
+separate post-processor to order qlog events in time if a tool do not provide this
+feature.
+
+There is no default base clock for formats that rely on a sharable clock. Tools
```

Comment:
```suggestion
There is no default base clock for formats that rely on a shareable clock. Tools
```

Reply:
This is also a bit vague... what exactly is a "shareable clock", terminology not used before this. Does it mean sharing the same clock across traces (e.g., one taken at the server, one at the client, later combined?) or does it mean deriving timing values from a single clock reading (i.e., relative and delta). I assume the latter, but the quiche PR that triggered this is mainly talking about the former?

Reply:
Yeah I'm not a fan of the word shareable here but struggled to find a suitable term. This is really just intended to be a rewording pf the old text that was focussed on unix-y time. If folks are using an abosulute timestamp, somewhere they should be indicating the actual format and rules for that, including where its epoch is anchored. Saying "you don't have to use UNIX time" like the old text is glossing over a lot of complexity.

Happy to rewrite this part if we can agree on what should be communicated.